crates/store/src/historical/mod.rs

```diff
@@ -0,0 +1,480 @@
+//! Historical tracking for `AccountTree` via mutation overlays
+
+use std::collections::VecDeque;
+use std::sync::{Arc, RwLock};
+
+use miden_objects::account::AccountId;
+use miden_objects::block::{AccountMutationSet, AccountTree, AccountWitness, BlockNumber};
+use miden_objects::crypto::merkle::{
+    LargeSmt,
+    LeafIndex,
+    MerkleError,
+    NodeIndex,
+    NodeMutation,
+    SMT_DEPTH,
+    SmtLeaf,
+    SmtStorage,
+    SparseMerklePath,
+};
+use miden_objects::{AccountTreeError, EMPTY_WORD, Word};
+
+/// Trait abstracting operations over different account tree backends.
+pub trait AccountTreeBackend {
```

Comment:
How is this different from the `AccountTreeBackend` that we have in `miden-base`? Do we need both?